App/src/main/java/cc/softwarefactory/lokki/android/MainActivity.java

```diff
@@ -277,22 +279,11 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     private void toggleVisibility() {
 
-        MainApplication.visible = !MainApplication.visible;
-        int visibility_mode = MainApplication.visible ? 0 : 1;
-        PreferenceUtils.setValue(this, PreferenceUtils.KEY_SETTING_VISIBILITY, String.valueOf(visibility_mode));
-
-        try {
-            if (MainApplication.visible) {
-                LocationService.start(MainActivity.this);
-                ServerAPI.setVisibility(MainActivity.this, true);
-                Toast.makeText(this, getResources().getString(R.string.you_are_visible), Toast.LENGTH_LONG).show();
-            } else {
-                LocationService.stop(MainActivity.this);
-                ServerAPI.setVisibility(MainActivity.this, false);
-                Toast.makeText(this, getResources().getString(R.string.you_are_invisible), Toast.LENGTH_LONG).show();
-            }
-        } catch (Exception ex) {
-            ex.printStackTrace();
+        Utils.setVisibility(!MainApplication.visible, MainActivity.this);
+        if (MainApplication.visible) {
+            Toast.makeText(this, getResources().getString(R.string.you_are_visible), Toast.LENGTH_LONG).show();
```

Comment:
I'm going to say that's better (but if you're convinced I'm wrong feel free to argue otherwise or get other people here to voice their opinion ;))